app/models/wiki.rb

```diff
@@ -1,3 +1,24 @@
 class Wiki < ActiveRecord::Base
   belongs_to :user
+
+  after_initialize :set_public
+
+  before_validation :disallow_private
+
+  t = Wiki.arel_table
+
+  scope :visible_to, -> (user) { user ? Wiki.where(t[:private].eq(false).or(t[:user_id].eq(user.id))) : Wiki.where(t[:private].eq(false)) }
```

Comment:
These are cool. I don't think you had to go down to the Arel level to do it (?), but the code is fine.